### PR TITLE
Fixes bug 1289899 - Add query date range to signature report.

### PR DESCRIPTION
**Pull request by adngdb**
I side-tracked a bit while working on this, and removed an old, unused function as well as improved a bit the look of the signature report search form.

### CODE REVIEW

File: webapp-django/crashstats/signature/jinja2/signature/signature_report.html [hunk 1]
```diff
@@ -13,21 +13,28 @@
 {% block content %}
 <div id="mainbody">
     <div class="page-heading">
+        <a href="#" class="display-toggle-filters toggle-filters show">Show Search Filters</a>
         <h2>Signature report for <em>{{ signature }}</em></h2>
+        <p>
+            Showing results from
+            <time class="ago" datetime="{{ query.start_date.isoformat() }}">{{ query.start_date | human_readable_iso_date }} UTC</time>
+            to <time class="ago" datetime="{{ query.end_date.isoformat() }}">{{ query.end_date | human_readable_iso_date }} UTC</time>.
+            <a href="#" class="toggle-filters">Edit filters</a>.
```
Comment (peterbe) [inline webapp-django/crashstats/signature/jinja2/signature/signature_report.html]: I don't like this. Sorry :)

It's confusing that it's labelled so differently from the "Show Search Filters". (ie. "edit" != "filter"). 
I think it's enough to have the "Show Search Filters" link in the upper right-hand corner.

File: webapp-django/crashstats/signature/jinja2/signature/signature_report.html [hunk 2]
```diff
@@ -13,21 +13,28 @@
 {% block content %}
 <div id="mainbody">
     <div class="page-heading">
+        <a href="#" class="display-toggle-filters toggle-filters show">Show Search Filters</a>
         <h2>Signature report for <em>{{ signature }}</em></h2>
+        <p>
+            Showing results from
+            <time class="ago" datetime="{{ query.start_date.isoformat() }}">{{ query.start_date | human_readable_iso_date }} UTC</time>
+            to <time class="ago" datetime="{{ query.end_date.isoformat() }}">{{ query.end_date | human_readable_iso_date }} UTC</time>.
```
Comment (peterbe) [inline webapp-django/crashstats/signature/jinja2/signature/signature_report.html]: We talked about it somewhere else but let's get that fancy new jinja template function here.

File: webapp-django/crashstats/signature/jinja2/signature/signature_report.html [hunk 3]
```diff
@@ -13,21 +13,28 @@
 {% block content %}
 <div id="mainbody">
     <div class="page-heading">
+        <a href="#" class="display-toggle-filters toggle-filters show">Show Search Filters</a>
```
Comment (peterbe) [inline webapp-django/crashstats/signature/jinja2/signature/signature_report.html]: This is an inline block preceding a block element. I think you should put this in a `<p>` tag and put `.display-toggle-filters` on it instead.

Comment (peterbe) [inline webapp-django/crashstats/signature/jinja2/signature/signature_report.html]: More importantly, can we make this a button instead?
It's a nit but it's odd that it's a hyperlink. Especially since you can't right-click and "Open in a New Window" on it. 

My webpages aren't perfect but I think the only time it's "OK" to have a hyperlink instead of a button is when it can actually be a link too. Like this:
```
<a href="{page_absolute_url}" onClick={showPermalinkWidget}>Permalink</a>
```

File: webapp-django/crashstats/signature/jinja2/signature/signature_report.html [hunk 4]
```diff
@@ -13,21 +13,28 @@
...
-        <a href="#" class="toggle-filters show">Show</a>
-        <h4>Search filters</h4>
-
         <form method="get" action="{{ url('signature:signature_report') }}"
             data-fields-url="{{ url('supersearch.search_fields') }}?exclude=signature"
             data-signature="{{ signature }}"
         >
             <button type="submit" id="search-button">Search</button>
-            <button class="new-line">new line</button>
+            <button class="new-line">New line</button>
```
Comment (peterbe) [inline webapp-django/crashstats/signature/jinja2/signature/signature_report.html]: Great!

File: webapp-django/crashstats/signature/jinja2/signature/signature_report.html [hunk 5]
```diff
@@ -13,21 +13,28 @@
...
 <div id="mainbody">
     <div class="page-heading">
+        <a href="#" class="display-toggle-filters toggle-filters show">Show Search Filters</a>
         <h2>Signature report for <em>{{ signature }}</em></h2>
+        <p>
+            Showing results from
+            <time class="ago" datetime="{{ query.start_date.isoformat() }}">{{ query.start_date | human_readable_iso_date }} UTC</time>
+            to <time class="ago" datetime="{{ query.end_date.isoformat() }}">{{ query.end_date | human_readable_iso_date }} UTC</time>.
+            <a href="#" class="toggle-filters">Edit filters</a>.
+        </p>
```
Comment (peterbe) [inline webapp-django/crashstats/signature/jinja2/signature/signature_report.html]: By the way, this looks great!!

File: webapp-django/crashstats/signature/views.py [hunk 6]
```diff
@@ -43,6 +43,43 @@
 )
 
 
+def get_date_range(params):
+    now = datetime.datetime.utcnow().replace(tzinfo=utc)
+
+    if not params.get('date'):
+        return (now - datetime.timedelta(days=7), now)
```
Comment (peterbe) [inline webapp-django/crashstats/signature/views.py]: Perhas instead of `7` we can make it `settings.DEFAULT_SIGNATURE_REPORT_RANGE_DAYS`?

File: webapp-django/crashstats/signature/views.py [hunk 7]
```diff
@@ -103,11 +140,12 @@ def signature_report(request, params, default_context=None):
...
-        utils.sanitize_dict(
-            get_report_list_parameters(params)
-        )
-    )
+    # Compute dates to show them to the user.
+    start_date, end_date = get_date_range(params)
+    context['query'] = {
+        'start_date': start_date,
+        'end_date': end_date,
+    }
```
Comment (peterbe) [inline webapp-django/crashstats/signature/views.py]: I love this!

File: webapp-django/crashstats/signature/static/signature/js/signature_report.js [hunk 8]
```diff
@@ -157,18 +157,16 @@ SignatureReport.init = function () {
         });
 
         // Show or hide filters.
-        searchSection.on('click', '.toggle-filters', function (e) {
+        $('.toggle-filters').click(function (e) {
```
Comment (peterbe) [inline webapp-django/crashstats/signature/static/signature/js/signature_report.js]: Ideally you should use the `.on('click'` instead. It's just a nit but I remember reading that all these hardcoded events lick `.click`, `.submit` are discouraged in favor of `.on('click', ` and `.on('submit', `

File: webapp-django/crashstats/signature/static/signature/js/signature_report.js [hunk 9]
```diff
@@ -157,18 +157,16 @@ SignatureReport.init = function () {
...
-            }
-            else {
-                elt.html('Hide');
-            }
+            searchSection.slideToggle(350);
+
+            // Update the main toggle link.
+            var mainToggleLink = $('.display-toggle-filters');
+            mainToggleLink.toggleClass('show');
+            mainToggleLink.html(
```
Comment (peterbe) [inline webapp-django/crashstats/signature/static/signature/js/signature_report.js]: Should be `.text(`.

File: webapp-django/crashstats/signature/static/signature/js/signature_report.js [hunk 10]
```diff
@@ -157,18 +157,16 @@ SignatureReport.init = function () {
...
+                (mainToggleLink.hasClass('show') ? 'Show' : 'Hide') + ' Search Filters'
```
Comment (peterbe) [inline webapp-django/crashstats/signature/static/signature/js/signature_report.js]: English in the js code isn't great. It means we're now spelling "Search Filters" in two places. 
(Generally, it's bad because it's hard to I18N static .js files but easy to do in the template rendering. However, that's not a concern in our case). 
I think you should change the HTML to be 
```html
<a href="#" class="display-toggle-filters toggle-filters show" 
  data-opposite="Hide Search Filters">Show Search Filters</a>
```
And then use:
```javascript
var opposite = mainToggleLink.data.opposite;
mainToggleLink.data.opposite = mainToggleLink.text();
mainToggleLink.text(opposite);
```

File: webapp-django/crashstats/signature/jinja2/signature/signature_report.html (hunk 3 above)
Comment (peterbe) [inline webapp-django/crashstats/signature/jinja2/signature/signature_report.html]: Also, I wonder about calling it "Search". **I** know it's a search because the underlying tech is SuperSearch. But the page is, after all, called "Signature report ..."
So perhaps "Show Filters" (and "Hide Filters").

File: webapp-django/crashstats/signature/views.py [hunk 11]
```diff
@@ -43,6 +43,43 @@
 )
 
 
+def get_date_range(params):
+    now = datetime.datetime.utcnow().replace(tzinfo=utc)
```
Comment (peterbe) [inline webapp-django/crashstats/signature/views.py]: Replace with `now = timezone.now()` perhaps?

**Comment by peterbe** [comment]
A couple of nits. I'm still a bit too tired to understand how this coincides with the (upcoming) changes in https://github.com/mozilla/socorro/pull/3421

**Comment by adngdb** [comment]
@peterbe ^^^ that's a pretty big commit. I found out that I had already made a function to get date boundaries from a set of parameters, so I removed the recent one, moved the old one to a new file, and moved over my unit tests. That was a good idea: the tests found a bug in the old function, so I fixed that (nothing impacting though, it was just an edge case that was incorrectly handled). 

I have also changed the layout of the Signature report page so that tabs look like actual tabs, and not buttons. I find it much better now. 

And of course, I addressed all your comments! :)

File: webapp-django/crashstats/topcrashers/tests/test_views.py [hunk 12]
```diff
@@ -41,15 +42,15 @@ def mocked_signature_first_date_get(**options):
...
+                            tzinfo=utc,
                         ),
                         "first_build": "20000101122334",
                     },
                     {
                         "signature": u"mozCool()",
                         "first_date": datetime.datetime(
                             2016, 5, 2, 0, 0, 0,
-                            tzinfo=UTC(),
+                            tzinfo=utc,
```
Comment (adngdb) [inline webapp-django/crashstats/topcrashers/tests/test_views.py]: Mmm, sorry, I made these changes while I was working on the ``get_date_boundaries`` function, but then I ended up moving it somewhere else, so it's weird that this file has any changes. Luckily, it's all very simple.

**Comment by peterbe** [comment]
There's a whitespace in front of the text "Show"
screen shot 2016-08-11 at 12 45 52 pm

File: webapp-django/crashstats/signature/jinja2/signature/signature_report.html [hunk 13]
```diff
@@ -14,20 +14,26 @@
...
-        <h4>Search filters</h4>
-
         <form method="get" action="{{ url('signature:signature_report') }}"
             data-fields-url="{{ url('supersearch.search_fields') }}?exclude=signature"
             data-signature="{{ signature }}"
         >
             <button type="submit" id="search-button">Search</button>
-            <button class="new-line">new line</button>
+            <button class="new-line">New line</button>
+            <h4>Search filters</h4>
```
Comment (peterbe) [inline webapp-django/crashstats/signature/jinja2/signature/signature_report.html]: "Search Filters" with a capital F.

**Comment by peterbe** [comment]
r+ with nits.

File: webapp-django/crashstats/crashstats/templatetags/jinja_helpers.py [hunk 14]
```diff
@@ -90,6 +90,17 @@ def timestamp_to_date(
...
+def time_ago(dt, format='%a, %b %d %H:%M %Z'):
+    return jinja2.Markup(
+        '<time datetime="{}" class="ago">{}</time>'
+        .format(
+            dt.isoformat(),
+            dt.strftime(format)
+        )
+    )
+
+
```
Comment (peterbe) [inline webapp-django/crashstats/crashstats/templatetags/jinja_helpers.py]: If we only ever generate `<time class="ago">` from django, how come `timeutils.js` is iterating over `$('time.in')`. Where do they get created?

Comment (adngdb) [inline webapp-django/crashstats/crashstats/templatetags/jinja_helpers.py]: Ah, I haven't checked that. I assumed we had some somewhere deep in the site, but maybe it is not used anymore.

Comment (adngdb) [inline webapp-django/crashstats/crashstats/templatetags/jinja_helpers.py]: Well I just checked and we of course use in the Tokens page, to indicate when a token will expire! :)

Comment (peterbe) [inline webapp-django/crashstats/crashstats/templatetags/jinja_helpers.py]: Oh. I was too quick to merge. 

Sorry.
